Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -262,36 +262,55 @@ private extension HTTPClient {
             return .failure(.unexpectedResponse(urlResponse))
         }
 
-        /// - Returns `nil` if status code is 304, since the response will be empty
-        /// and fetched from the eTag.
-        func dataIfAvailable(_ statusCode: HTTPStatusCode) -> Data? {
-            if statusCode == .notModified {
-                return nil
-            } else {
-                return data
-            }
-        }
+        let statusCode: HTTPStatusCode = .init(rawValue: httpURLResponse.statusCode)
 
-        let statusCode = HTTPStatusCode(rawValue: httpURLResponse.statusCode)
+        // `nil` if status code is 304, since the response will be empty and fetched from the eTag.
+        let dataIfAvailable = statusCode == .notModified
+            ? nil
+            : data
 
+        return self.createVerifiedResponse(request: request,
+                                           urlRequest: urlRequest,
+                                           data: dataIfAvailable,
+                                           response: httpURLResponse)
+    }
+
+    /// - Returns `Result<VerifiedHTTPResponse<Data>, NetworkError>?`
+    private func createVerifiedResponse(
+        request: Request,
+        urlRequest: URLRequest,
+        data: Data?,
+        response httpURLResponse: HTTPURLResponse
+    ) -> VerifiedHTTPResponse<Data>.Result? {
         return Result
-            .success(dataIfAvailable(statusCode))
-            .mapToResponse(response: httpURLResponse,
-                           request: request.httpRequest,
-                           signing: self.signing(for: request.httpRequest),
-                           verificationMode: request.verificationMode)
+            .success(data)
+            .mapToResponse(response: httpURLResponse, request: request.httpRequest)
+            // Fetch from ETagManager if available
             .map { (response) -> HTTPResponse<Data>? in
-                guard let cachedResponse = self.eTagManager.httpResultFromCacheOrBackend(
+                return self.eTagManager.httpResultFromCacheOrBackend(
```

Comment:
> This might be ok though... if we receive a 304 on a request that already failed to verify, it makes sense that it still fails to verify

Yeah 👍🏻 
Although as discussed this has to change again.